lightning/src/ln/outbound_payment.rs

```diff
@@ -2185,7 +2238,9 @@ mod tests {
 		assert!(!outbound_payments.has_pending_payments());
 		assert!(pending_events.lock().unwrap().is_empty());
 
-		assert!(outbound_payments.add_new_awaiting_invoice(payment_id, Retry::Attempts(0)).is_ok());
+		assert!(
+			outbound_payments.add_new_awaiting_invoice(payment_id, Retry::Attempts(0), None).is_ok()
```

Comment:
Could we test that a`Some` value here makes it into the `RouteParameters` above?